libsolidity/inlineasm/AsmAnalysis.cpp

```diff
@@ -202,13 +202,12 @@ bool AsmAnalyzer::operator()(assembly::VariableDeclaration const& _varDecl)
 
 bool AsmAnalyzer::operator()(assembly::FunctionDefinition const& _funDef)
 {
-	Scope& bodyScope = scope(&_funDef.body);
+	Scope& varScope = scope(m_info.virtualBlocks.at(&_funDef).get());
```

Comment:
it is called a virtual block because it does not have a syntactic equivalent
we can add an assert, but I don't see much difference between `solAssert` and `at()`. Yes, it throws a different exception, but does that really matter so much?
Or do you mean that `get()` does not return `nullptr`?